test/test_testing.py

```diff
@@ -1302,6 +1328,56 @@ def test_config_from_directory(self):
         self.assertIsNone(harness._backend._config._defaults['opt_null'])
         self.assertIsNone(harness._backend._config._defaults['opt_no_default'])
 
+    def test_config_from_directory_charmcraft_yaml(self):
+        tmp = pathlib.Path(tempfile.mkdtemp())
+        self.addCleanup(shutil.rmtree, str(tmp))
+        charmcraft_filename = tmp / 'charmcraft.yaml'
+        with charmcraft_filename.open('wt') as charmcraft_yaml:
+            charmcraft_yaml.write(textwrap.dedent('''
+            type: charm
+            bases:
+              - build-on:
+                - name: ubuntu
+                  channel: "22.04"
+                run-on:
+                - name: ubuntu
+                  channel: "22.04"
+
+            config:
+                options:
+                    opt_str:
```

Comment:
We don't need so many options to test this -- we're just testing that it gets loaded from the right place. I think two would be fine, maybe `opt_str` and `opt_int`.